packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -9,8 +9,45 @@ import Module from 'module';
 import {join} from 'path';
 import importFresh from 'import-fresh';
 import {LoadContext, Plugin, PluginConfig} from '@docusaurus/types';
+import {Schema} from 'yup';
+const chalk = require('chalk');
 import {CONFIG_FILE_NAME} from '../../constants';
 
+function printError(error) {
+  console.log(
+    chalk.red(
+      `Validation Errors:${error?.errors?.reduce(
+        (formatedError, error, i) => `${formatedError}\n${i + 1}. ${error}`,
+        '',
+      )}`,
+    ),
+  );
+  process.exit(1);
+}
+
+function validate<T>(schema: Schema<T>, options: unknown) {
+  try {
+    return schema.validateSync(options, {
+      abortEarly: false,
+    });
+  } catch (error) {
+    printError(error);
+    return;
+  }
+}
+
+function validateAndStrip<T>(schema: Schema<T>, options: unknown) {
+  try {
+    return schema.validateSync(options, {
+      abortEarly: false,
+      stripUnknown: true,
```

Comment:
not very fan of this option. If user does a typo in a config key, and the config is not used, he might not notice. Having an error in such case is not a bad idea